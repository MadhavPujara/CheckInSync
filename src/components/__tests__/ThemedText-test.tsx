import * as React from "react";
import renderer, { type ReactTestRenderer } from "react-test-renderer";
import ThemedText from "@/components/ThemedText";
import { useColorScheme } from "react-native";

// Mock the theme related imports
jest.mock("@/theme/index", () => ({
    __esModule: true,
    Colors: {
        light: {
            text: "#000000",
            background: "#ffffff",
        },
        dark: {
            text: "#ffffff",
            background: "#000000",
        },
    },
    default: {
        Colors: {
            light: {
                text: "#000000",
                background: "#ffffff",
            },
            dark: {
                text: "#ffffff",
                background: "#000000",
            },
        },
    },
}));

// Mock the useThemeColor hook to return predictable colors
jest.mock("@/hooks/useThemeColor", () => ({
    __esModule: true,
    default: jest.fn().mockImplementation((props, colorName) => {
        return props?.light || "#000000";
    }),
}));

// Mock the useColorScheme hook with a variable we can change
const mockColorScheme = jest.fn().mockReturnValue("light");
jest.mock("react-native", () => ({
    ...jest.requireActual("react-native"),
    useColorScheme: () => mockColorScheme(),
}));

describe("ThemedText", () => {
    let testRenderer: ReactTestRenderer;

    afterEach(() => {
        testRenderer?.unmount();
        jest.clearAllMocks();
    });

    it("renders default text correctly", () => {
        testRenderer = renderer.create(<ThemedText>Default text</ThemedText>);
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });

    it("renders title text correctly", () => {
        testRenderer = renderer.create(
            <ThemedText type="title">Title text</ThemedText>
        );
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });

    it("renders subtitle text correctly", () => {
        testRenderer = renderer.create(
            <ThemedText type="subtitle">Subtitle text</ThemedText>
        );
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });

    it("renders link text correctly", () => {
        testRenderer = renderer.create(
            <ThemedText type="link">Link text</ThemedText>
        );
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });

    it("applies custom colors correctly", () => {
        testRenderer = renderer.create(
            <ThemedText lightColor="#ff0000" darkColor="#00ff00">
                Colored text
            </ThemedText>
        );
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });

    it("respects dark theme", () => {
        // Update the mock for this test
        mockColorScheme.mockReturnValue("dark");

        // Also update the useThemeColor mock for dark theme
        const useThemeColor = require("@/hooks/useThemeColor").default;
        useThemeColor.mockImplementationOnce(
            (props: { light?: string; dark?: string }, colorName: string) => {
                return props?.dark || "#ffffff";
            }
        );

        testRenderer = renderer.create(
            <ThemedText>Dark theme text</ThemedText>
        );
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });
});
