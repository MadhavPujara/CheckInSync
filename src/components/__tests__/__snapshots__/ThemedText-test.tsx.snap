// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedText applies custom colors correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#ff0000",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Colored text
</Text>
`;

exports[`ThemedText renders default text correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Default text
</Text>
`;

exports[`ThemedText renders link text correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      {
        "color": "#0a7ea4",
        "fontSize": 16,
        "lineHeight": 30,
      },
      undefined,
    ]
  }
>
  Link text
</Text>
`;

exports[`ThemedText renders subtitle text correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
      },
      undefined,
      undefined,
      undefined,
      {
        "fontSize": 20,
        "fontWeight": "bold",
      },
      undefined,
      undefined,
    ]
  }
>
  Subtitle text
</Text>
`;

exports[`ThemedText renders title text correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
      },
      undefined,
      {
        "fontSize": 32,
        "fontWeight": "bold",
        "lineHeight": 32,
      },
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Title text
</Text>
`;

exports[`ThemedText respects dark theme 1`] = `
<Text
  style={
    [
      {
        "color": "#ffffff",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Dark theme text
</Text>
`;
